### ISSUE TITLE
reactor.netty.http.client.PrematureCloseException: Connection prematurely closed DURING response

### ISSUE THREAD

**Issue by trivedishalini** [reactor/reactor-netty#1876]
Getting issue with reactor.netty while forwarding request to backend services. Even though backed service sending back response but reactor.netty prematurely closed connection which causing failure of services.

**Below are version details** 

Spring Boot version: 2.1.4.RELEASE
 Reactor core versoin: 3.3.5.RELEASE 
Reactor netty : 0.9.7.RELEASE
Reactor extra : 3.3.3.RELEASE
java8
windows

**Error in logs**

```
reactor.core.Exceptions$ReactiveException: reactor.netty.http.client.PrematureCloseException: Connection prematurely closed DURING response
        at reactor.core.Exceptions.propagate(Exceptions.java:326)
        at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:91)
        at reactor.core.publisher.Mono.block(Mono.java:1494
        at org.springframework.boot.actuate.health.HealthIndicatorReactiveAdapter.invoke(HealthIndicatorReactiveAdapter.java:48)
        at org.springframework.boot.actuate.health.HealthIndicatorReactiveAdapter.lambda$null$0(HealthIndicatorReactiveAdapter.java:43)
        at reactor.core.scheduler.ElasticScheduler$DirectScheduleTask.run(ElasticScheduler.java:291)
        at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:50)
        at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:27)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
        Suppresseed: java.lang.Exception: #block terminated with an error
                at reactor.cor.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:93)
                ... 13 more
```

 **Steps to Reproduce**
issue observed during performance testing when more that 10 request iteration perform. Issue observed while polling of the request and parallel execution too. Single iteration work successfully for same request.

 **Possible Solution**
Below PR could be one solution for this issue.
https://github.com/reactor/reactor-netty/pull/1442

 **Environment**
Windows environment with  3 node kubernetes cluster.

**Comment by violetagg** [comment]
@trivedishalini You components' versions are too old. Please upgrade and test again.

> Possible Solution
> Below PR could be one solution for this issue.
> #1442

In the mentioned PR (that is rejected) there is a link to a fix in Spring Cloud Gateway. The fix is available in Spring Cloud Gateway v2.2.7.RELEASE onwards.
https://github.com/reactor/reactor-netty/pull/1442#issuecomment-752662290
https://github.com/spring-cloud/spring-cloud-gateway/pull/2093
https://github.com/spring-cloud/spring-cloud-gateway/commit/2f2163c49a6e00db62d26290b2a8909083e7143e

I've edited your comment to improve the formatting. You might want to check out this [Mastering Markdown guide](https://guides.github.com/features/mastering-markdown/) for future reference.

**Comment by violetagg** [comment]
@trivedishalini Were you able to upgrade and test?

**Comment by jiangxiaoqiang** [comment]
I am facing the same problem with spring boot 2.3.10.RELEASE

**Comment by violetagg** [comment]
@jiangxiaoqiang Please open a new issue where we will track your particular use case. Please provide reproducible example. Thanks!

**Comment by violetagg** [comment]
@@trivedishalini I'm closing this one. If you have the issue after the upgrade, we can reopen it.

**Comment by echozhangxiaomin** [comment]
我也遇见了这个问题。

**Comment by trivedishalini** [comment]
@violetagg , issue resolved after upgrading spring boot 2.5 ,netty-codec 4.1.69.Final, reactor-netty 1.0.12 , reactor-core 3.4.11 . 
thanks for suggestion.

**Comment by chgang** [comment]
spring-boot: 2.6.15
reactor-netty:1.0.23
reactor-core:3.4.29
I am facing the same problem .
